org.eclipse.xtend.core/src/org/eclipse/xtend/core/compiler/XtendCompilerHelper.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2023 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtend.core.compiler;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.xtend.core.xtend.AnonymousClass;
+import org.eclipse.xtend.core.xtend.XtendField;
+import org.eclipse.xtend.core.xtend.XtendFunction;
+import org.eclipse.xtend.core.xtend.XtendMember;
+import org.eclipse.xtext.common.types.JvmType;
+import org.eclipse.xtext.util.IResourceScopeCache;
+import org.eclipse.xtext.util.Tuples;
+import org.eclipse.xtext.xbase.jvmmodel.IJvmModelAssociations;
+
+import com.google.inject.Inject;
+
+/**
+ * @author Lorenzo Bettini - Initial contribution and API
+ */
+public class XtendCompilerHelper {
```

Comment:
Since this can only answer whether a JvmType can be compiled to an anonymous class, maybe a more restrictive name should be chosen?
`AnonymousClassCompilerHelper`?

Reply:
Indeed that was my second choice, so I'll rename it :)

Reply:
By the way, we already have https://github.com/eclipse/xtext/blob/88acf1555988cd81e1756abc0d6d2c3f1f1efa11/org.eclipse.xtend.core/src/org/eclipse/xtend/core/jvmmodel/AnonymousClassUtil.java#L39
should I put the methods of `XtendCompilerHelper` directly there?